install.sh

```diff
@@ -1,62 +1,167 @@
 #!/bin/bash
 
-if [[ -e /etc/debian_version ]] || [[ -e /etc/fedora-release ]] || [[ -e  /etc/centos-release ]] || [[ -e /etc/arch-release ]]; then
-	:
-else
-	echo "Looks like you aren't running this installer on a Debian, Ubuntu, Fedora, CentOS, or Arch Linux system."
-	exit 1
-fi
+clear
+# Clear screen before running any commands
 
+REQUIREMENTS_INSTALLED=0
 
-command -v jq >/dev/null 2>&1 && { JQ=0; }
-#install python
-if ! command -v python3 &> /dev/null; then
-	if [[ -e /etc/debian_version ]]; then
-		sudo apt install python3
-	elif [[ -e /etc/fedora-release ]]; then
-		sudo dnf install python3
-	elif [[ -e /etc/centos-release ]]; then
-		sudo yum install -y python3
-	elif [[ -e /etc/arch-release ]]; then
-		sudo pacman -S python3
-	fi
-else
-	echo "Skipping installing python, as python is preinstalled."
-fi
-#install tkinter, a dependency
-#install jq, needed to get version number
-if [[ -e /etc/debian_version ]]; then
-	sudo apt install python3-tk jq python3-pip
-elif [[ -e /etc/fedora-release ]]; then
-	sudo dnf install python3-tkinter jq python3-pip
-elif [[ -e /etc/centos-release ]]; then
-	rpm -q epel-release &> /dev/null || EPEL=0 
-	sudo yum install -y python3-tkinter epel-release python3-pip
-	sudo yum install -y jq
-	[[ $EPEL -eq 0 ]] && sudo yum remove -y epel-release
-elif [[ -e /etc/arch-release ]]; then
-	sudo pacman -S --needed tk jq
-fi
+install_fail () {
+    echo "Install Failed."
+    exit 1
+}
+
+install_debian () {
+    sudo apt install python3 python3-tk python3-pip git || install_fail
+}
+
+install_fedora () {
+    sudo dnf install python3 python3-tkinter python3-pip git || install_fail
+}
+
+install_centos () {
+    sudo yum install -y python3 || install_fail
+    rpm -q epel-release &> /dev/null || EPEL=0
+    sudo yum install -y python3-tkinter epel-release python3-pip git || install_fail
+    # Honestly not sure why it's installing epel and then uninstalling
+    [[ $EPEL -eq 0 ]] && sudo yum remove -y epel-release
+}
+
+install_arch () {
+    sudo pacman -S --needed python3 tk git && python3 -m ensurepip || install_fail
+}
+
+install_macos() {
+    echo "This script will install homebrew, if you do not wish to install homebrew, exit within 5 seconds..."
+    sleep 5
+    if test ! $(which brew); then
```

Comment:
Quote this to prevent word splitting.
```suggestion
    if test ! "$(which brew)"; then
```

See for more details: https://github.com/koalaman/shellcheck/wiki/SC2046